mars/optimizes/utils.py

```diff
@@ -0,0 +1,73 @@
+class Optimizer:
+    def __init__(self, graph, engine):
+        self.engine = engine
+        self.explored = set()
+        self.graph = graph
+        self.keys = []
+
+    # this is method is used to be compatible with jax.numpy
+    @staticmethod
+    def _jax_compat(op):
+        if hasattr(op, 'execute_jax'):
+            try:
+                op.execute_jax()
+            except NotImplementedError:
+                return False
+            return True
+        return False
```

Comment:
For those not supported operand, return `NotImplementedError` so that we don't need to check by calling them.